judgels-frontends/raphael/src/routes/uriel/contests/single/overview/ContestRegistrationCard/ContestRegistrationCard.tsx

```diff
@@ -139,6 +144,21 @@ class ContestRegistrationCard extends React.PureComponent<ContestRegistrationCar
     return <ContestRegistrantsDialog onClose={this.toggleRegistrantsDialog} />;
   };
 
+  private renderRegistrationConfirmationDialog = () => {
```

Comment:
OK, a better idea:

If `termsAndConditions` is `undefined` or `termsAndConditions.contest` is undefined, this should immediately register the contest. So that it is backward compatible, and/or we can turn off this feature if not needed.

If not, then show the confirmation dialog.